### PR TITLE
vsock

### PR DESCRIPTION
**Pull request by jsorg71**
Always have the os_calls vsock functions defined.   More consistent with other function.
Print an error if vsock is requested but not compiled.

### CODE REVIEW

File: common/os_calls.c [hunk 1]
```diff
@@ -990,6 +1004,24 @@ g_sck_local_bind(int sck, const char *port)
...
 }
 
+/*****************************************************************************/
+int
+g_sck_vsock_bind(int sck, const char *port)
+{
+#if defined(XRDP_ENABLE_VSOCK)
+    struct sockaddr_vm s;
+
+    memset(&s, 0, sizeof(struct sockaddr_vm));
```
Comment (metalefty) [inline common/os_calls.c]: g_memset?

**Comment by metalefty** [comment]
Maybe I should comment on Jay's change here. Please forget.

File: common/os_calls.c [hunk 2]
```diff
@@ -583,6 +586,17 @@ g_sck_local_socket(void)
...
 
 /*****************************************************************************/
+int
+g_sck_vsock_socket(void)
+{
+#if defined(XRDP_ENABLE_VSOCK)
+    return socket(PF_VSOCK, SOCK_STREAM, 0);
+#else
+    return 0;
+#endif
```
Comment (hardening) [inline common/os_calls.c]: should be -1

Comment (jsorg71) [reply common/os_calls.c]: Yea, xrdp currently treats sockets < 1 as invalid.  This should be changed in many places.  I'll add an issue for this.

Comment (jsorg71) [reply common/os_calls.c]: I add #938 for this

**Pull request merged**